### PR TITLE
Dynamic Ingestion of a Website

### PR DESCRIPTION
**Pull request by sukantadash**
https://issues.redhat.com/browse/REDSAIA-107

Tested by running below pipeline
https://console-openshift-console.apps.rhsaia.vg6c.p1.openshiftapps.com/k8s/ns/rhsaia-v2-dev/tekton.dev~v1~PipelineRun/website-ingestion-pipeline-0fho8l

### CODE REVIEW

File: kfp/website-ingestor/ingestion-pipeline-website-local.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+        exit(1)
+
+    auth_config = weaviate.auth.AuthApiKey(api_key=weaviate_api_key)
+    return weaviate.Client(
+        url=f"{weaviate_host}:{weaviate_port}",
+        auth_client_secret=auth_config,
+    )
+
+
+def get_top_records(index_name, limit=10):
```
Comment (jland-redhat) [inline kfp/website-ingestor/ingestion-pipeline-website-local.py:32]: Are we using this for anything or just testing?

File: kfp/website-ingestor/ingestion-pipeline-website-local.py [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+            logger.error(f"Error during ingestion for index {index_name}: {e}")
+
+
+    """Process the website content and add it to the Weaviate store."""
+    logger.info(f"Starting processing for URL: {url}")
+    weaviate_client = get_weaviate_client()
+
+
+    # Ensure the index (class) exists
+    index_name = "WebScrapedData"
```
Comment (jland-redhat) [inline kfp/website-ingestor/ingestion-pipeline-website-local.py]: Can we make this dynamic and also something that gets passed into the pipeline by tekton

**Review by jland-redhat — CHANGES_REQUESTED** [review]
The biggie is just making the index we put this in dynamic

File: kfp/website-ingestor/ingestion-pipeline-website-local.py (hunk 1 above)
Comment (sukantadash) [reply kfp/website-ingestor/ingestion-pipeline-website-local.py:32]: This is only for testing

File: kfp/website-ingestor/ingestion-pipeline-website-local.py (hunk 2 above)
Comment (sukantadash) [reply kfp/website-ingestor/ingestion-pipeline-website-local.py]: its dynamic, we can pass website url in the pipeline run

Comment (sukantadash) [reply kfp/website-ingestor/ingestion-pipeline-website-local.py]: Added the vectordb index as a pipeline parameter

**Review by jland-redhat — APPROVED** [review]

**Review by jland-redhat — APPROVED** [review]

**Pull request merged**